File: packages/llm-test-harness/src/metrics/exact-match.ts
import type { MetricFn, MetricScore } from '../types.js'

export class ExactMatch implements MetricFn {
  readonly name = 'ExactMatch'

  constructor(
    private expected: string,
    private caseSensitive = true,
  ) {}

  evaluate(text: string): MetricScore {
    const a = this.caseSensitive ? text : text.toLowerCase()
    const b = this.caseSensitive ? this.expected : this.expected.toLowerCase()
    const pass = a === b
    return { name: this.name, pass, score: pass ? 1 : 0 }
  }
}

File: packages/llm-test-harness-python/llm_test_harness/metrics/exact_match.py
from __future__ import annotations

from ..types import MetricFn, MetricScore


class ExactMatch(MetricFn):
    def __init__(self, expected: str) -> None:
        self._expected = expected

    @property
    def name(self) -> str:
        return "ExactMatch"

    def evaluate(self, text: str) -> MetricScore:
        passed = text == self._expected
        return MetricScore(
            name=self.name,
            passed=passed,
            score=1.0 if passed else 0.0,
        )
